internal/bminventory/inventory.go

```diff
@@ -392,6 +398,68 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 	return installer.NewRegisterClusterCreated().WithPayload(&cluster.Cluster)
 }
 
+func (b *bareMetalInventory) RegisterDay2Cluster(ctx context.Context, params installer.RegisterDay2ClusterParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	id := params.NewDay2ClusterParams.ID
+	url := installer.GetClusterURL{ClusterID: *id}
+	consoleURL := swag.StringValue(params.NewDay2ClusterParams.APIVipDnsname)
+	clusterName := swag.StringValue(params.NewDay2ClusterParams.Name)
+	openshiftVersion := swag.StringValue(params.NewDay2ClusterParams.OpenshiftVersion)
+
+	log.Infof("Register day2-cluster: %s with id %s", clusterName, id.String())
+
+	cluster := common.Cluster{Cluster: models.Cluster{
+		ID:               id,
+		Href:             swag.String(url.String()),
+		Kind:             swag.String(models.ClusterKindClusterDay2),
+		Name:             clusterName,
+		OpenshiftVersion: openshiftVersion,
+		UpdatedAt:        strfmt.DateTime{},
+	}}
+
+	err := validations.ValidateClusterNameFormat(clusterName)
+	if err != nil {
+		return common.NewApiError(http.StatusBadRequest, err)
+	}
+
+	// Persist worker-ignition to s3 for cluster
+	ignitionConfig, err := b.formatNodeIgnitionFile(consoleURL)
+	if err != nil {
+		log.WithError(err).Errorf("failed to format ignition config file for cluster %s", cluster.ID)
+	}
+	fileName := fmt.Sprintf("%s/%s", cluster.ID, workerIgnition)
+	err = b.objectHandler.Upload(ctx, ignitionConfig, fileName)
+	if err != nil {
+		return installer.NewRegisterDay2ClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, fmt.Errorf("failed to upload %s to s3", fileName)))
+	}
+
+	// After registering the cluster, its status should be 'ClusterStatusDay2cluster'
+	err = b.clusterApi.RegisterDay2Cluster(ctx, &cluster)
+	if err != nil {
+		log.Errorf("failed to register cluster %s ", clusterName)
+		return installer.NewRegisterDay2ClusterInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	return installer.NewRegisterDay2ClusterCreated().WithPayload(&cluster.Cluster)
+}
+
+func (b *bareMetalInventory) formatNodeIgnitionFile(consoleURL string) ([]byte, error) {
+	var ignitionParams = map[string]string{
+		"SOURCE": "http://" + consoleURL + ":22624/config/worker",
```

Comment:
consoleURL can be renamed now... maybe apiAddress or hostname.

Reply:
fixed